libs/components/src/popover/vl-floating-ui.controller.ts

```diff
@@ -0,0 +1,195 @@
+import { LitElement, ReactiveController } from 'lit';
+import { computePosition, arrow, flip, shift, offset, type Placement, Middleware } from '@floating-ui/dom';
+
+type FloatingControllerOptions = {
+    reference: string;
+    trigger: string;
+    placement: Placement;
+    distance: number;
+    showDelay: number;
+    hideDelay: number;
+};
+
+interface FloatingElement extends LitElement {
+    open: boolean;
+    hide: () => void;
+    show: () => void;
+    toggle: () => void;
+}
+
+export default class FloatingController implements ReactiveController {
+    private host: FloatingElement;
+    private hoverTimeout = 0;
+    private options!: FloatingControllerOptions;
+
+    constructor(host: FloatingElement, options: FloatingControllerOptions) {
+        this.setOptions(options);
+        (this.host = host).addController(this);
+    }
+
+    hostConnected() {
+        this.handleClickOutside = this.handleClickOutside.bind(this);
+        this.handleClick = this.handleClick.bind(this);
+        this.handleKeyDown = this.handleKeyDown.bind(this);
+        this.handleMouseOver = this.handleMouseOver.bind(this);
+        this.handleMouseOut = this.handleMouseOut.bind(this);
+        this.handleFocusIn = this.handleFocusIn.bind(this);
+        this.handleFocusOut = this.handleFocusOut.bind(this);
+        this.handleResize = this.handleResize.bind(this);
+        this.addEventListeners();
+    }
+
+    hostDisconnected() {
+        this.removeEventListeners();
+    }
+
+    get referenceElement(): HTMLElement {
```

Comment:
Ik ben geen grote fan van deze stijl van getters, @kspeltix willen we dit blijven hanteren?

Reply:
Ik gebruik dat persoonlijk nooit, maar ik heb dan ook een Java achtergrond.

Ik heb er nooit nood aan, mijn best practices zijn:
- een voorkeur voor functie aanroepen, zeker voor iets wat een functie is en niet zomaar een getter of een setter (dit geval)
- als je er gewoon een member mee ontsluit is het nutteloos, gebruik dan de member rechtstreeks
- als het is om een 'neveneffect' te introduceren, doe het niet of noem het dan anders

Reply:
aangepast